### PR TITLE
Update LangVersion and enable implicit usings

### PR DESCRIPTION
**Pull request by j3parker**

### CODE REVIEW

File: Directory.build.props [hunk 1]
```diff
@@ -1,6 +1,7 @@
 <Project>
 	<PropertyGroup>
 		<Deterministic>true</Deterministic>
-		<LangVersion>9.0</LangVersion>
+		<LangVersion>10.0</LangVersion>
+		<ImplicitUsings>true</ImplicitUsings>
```
Comment (j3parker) [inline Directory.build.props:5]: This is the default for new projects in their template. It maybe feels weird but I think its pretty nice when you get used to it. The list depends on your project's SDK, for normal .NET things it is:

```
System
System.Collections.Generic
System.IO
System.Linq
System.Net.Http
System.Threading
System.Threading.Tasks
```

File: src/D2L.CodeStyle.Annotations/D2L.CodeStyle.Annotations.csproj [hunk 2]
```diff
@@ -2,7 +2,8 @@
 
   <PropertyGroup>
     <TargetFrameworks>net20;netstandard2.0</TargetFrameworks>
-    <LangVersion>7.3</LangVersion>
+    <!-- We build this for .NET 2.0 and the implicit usings bring in namespaces that didn't exist back then -->
+    <ImplicitUsings>disable</ImplicitUsings>
```
Comment (j3parker) [inline src/D2L.CodeStyle.Annotations/D2L.CodeStyle.Annotations.csproj:6]: .NET 2.0 was for the COM stuff, I'm not sure if we still need to keep doing this but whatever, its just the annotations DLL.

File: Directory.build.props (hunk 1 above)
Comment (j3parker) [reply Directory.build.props:5]: `<nullable>` is also enabled by default. I haven't done that yet that in this PR because there will be some clean-up.

**Review by omsmith — APPROVED** [review]

File: Directory.build.props (hunk 1 above)
Comment (JeffAshton) [reply Directory.build.props:5]: The `ImplicitUsings` scares me, but not enough to get use to the new world

**Review by JeffAshton — APPROVED** [review]

File: Directory.build.props (hunk 1 above)
Comment (j3parker) [reply Directory.build.props:5]: It's pretty questionable. Like they better be really careful about adding new things in those namespaces... but 99% of the time you end up having all of those anyway so it's really not much of a change.

Comment (j3parker) [reply Directory.build.props:5]: Mostly I just don't want to fight it, but it does make writing a new file a little less annoying (which is a niche concern)

**Pull request merged**